setup.py

```diff
@@ -410,6 +470,22 @@ def get_hdf5_version(headername):
         # Don't delete this argument. It maybe useful for distutils
         # when adding more flags later on
         # sys.argv.remove(arg)
+    elif arg.find('--use-pkgconfig') == 0:
+        if arg == '--use-pkgconfig':
+            USE_PKGCONFIG = True
+        else:
+            USE_PKGCONFIG = arg.split('=')[1]
+
+
+if USE_PKGCONFIG is None:
+    USE_PKGCONFIG = os.environ.get('USE_PKGCONFIG', 'TRUE')
+if isinstance(USE_PKGCONFIG, str):
```

Comment:
My fear is that pkg-config can interfere with other options specified by the user (--hdf5, etc.).
At a first look it should not but the code is quite complex and I'm not sure I see all paths.
This is mainly a conservative choice.
I can make some additional test and enable pkg-config by default always.